packages/yew/src/dom_bundle/bcomp/scope.rs

```diff
@@ -1,17 +1,14 @@
 //! Component scope module
```

Comment:
Actually, I think the scope and lifecycle runners should not be put inside of `dom_bundle`.

If we were to produce a feature gate that disables client side rendering logic for server side renderer and these types are not in the `dom_bundle`, then one can simply feature gate on `mod dom_bundle`.

Reply:
I agree. Component related code should be outside of `dom_bundle`

Reply:
To actually be able to feature gate `dom_bundle` (I think that is a valid goal) one would have to move/replace `btag::{InputFields, Value}`. `bcomp::{Mountable, PropsWrapper}` would get a replacement by a `ssr` version (currently `render_to_string` is part of it, but a client `Mountable` with `mount` and a server `Mountable` with `render_to_string` would make sense). The `AppHandle` and `set_event_bubbling` shouldn't need to get used on the server anyway.

Basically what I envision for this toggle is that there is client `dom_bundle` and server `yeet_bundle` that offer the same `pub(crate)` interface and get toggled between with `feature = "ssr"`, if I understood you correctly @futursolo

Reply:
> btag::{InputFields, Value}. bcomp::{Mountable, PropsWrapper} would get a replacement by a ssr version

I think it's just the `Apply` on `InputFields` and `Value` makes them dom-dependent.
I guess we can leave the `Apply` trait in `dom_bundle` and move `InputFields` and `Value`?

`Mountable` and `PropsWrapper` is also used by `VComp` which means that if it's shared by csr and ssr, then these 2 features can co-exist (Not saying that they are useful when both enabled, but things like `cargo clippy --all-features` can make development easier).

If a different variant is used here, then `csr` and `ssr` becomes mutually exclusive, which I think is a bigger downside.

As for why there isn't a `yeet_bundle`, it's due to that the "rendered" form of a VNode on the server side is a `String`.
(There's no 2nd render, hence no need for reconciliation.
This is necessary so that we can reliably hydrate the first successfully rendered `Html` on the client side.) I think creating a `Y*` system in which its sole purpose is to call a single `render_to_string()` is like cracking a nut with a sledgehammer.

We can consider moving `render_to_string` to a trait, but this requires boxing.

Reply:
We might want to add an ssg feature in the future for generation static files. I believe that could be the current `render_to_string` but instead it renders each and every page and saves it as a file.

Not sure how much we need to account for that in the design today, unless the SSR work that is done by @fursolo can be trivially adapted

Reply:
I don't think `ssg` should be provided as part of the `yew` crate.

It can be delivered with the public api of `ServerRenderer` and it requires a framework-ish support (Next.js or Gatsby) for it to work (cli + a certain directory structure).

Reply:
I don't think we can have a certain directory structure because of Rust. 

It would be more like an enum of pages where each variant is mapped to a component and may optionally provide props to be passed to the component. It could be part of yew crate or could be a new crate, `yew-ssg`.

If we're going for different crates, we could have a `yew-core` crate which has all the library code. The `yew` crate would re-export `yew-core`, `yew-router`, etc. That way we can have `yew-ssg` which implements static site generation, `yew-mobile`/`yew-desktop` could be a thing too.